compiler/quilt/nodes.py

```diff
@@ -79,11 +79,21 @@ def __setattr__(self, name, value):
 
     def __repr__(self):
         pinfo = super(GroupNode, self).__repr__()
-        group_info = '\n'.join(name + '/' for name in sorted(self._group_keys()))
-        if group_info:
-            group_info += '\n'
-        data_info = '\n'.join(sorted(self._data_keys()))
-        return '%s\n%s%s' % (pinfo, group_info, data_info)
+        items = [name + '/' for name in sorted(self._group_keys())]
+        if not items:
```

Comment:
I think you'll want to remove this "if not items". With that in place, it skips the data nodes for top-level packages that contain only data nodes (not any group nodes).